dbms/src/Flash/Mpp/MPPTaskManager.cpp

```diff
@@ -309,4 +347,58 @@ void MPPTaskManager::releaseThreadsFromScheduler(const int needed_threads)
     scheduler->releaseThreadsThenSchedule(needed_threads, *this);
 }
 
+void MPPTaskManager::monitorMPPTasks()
+{
+    std::unique_lock lock(mu, std::defer_lock);
+    while (true)
+    {
+        lock.lock();
```

Comment:
I think just hold the lock and get a snapshot of current monitored_tasks is enough. We can do the monitor things based on the snapshot.